inbox/models/message.py

```diff
@@ -340,6 +340,21 @@ def create_from_synced(cls, account, mid, folder_name, received_date, body_strin
                         mid=mid,
                     )
                     msg._mark_error()
+                finally:
+                    # Even though whole email can be only 16MB in total, it can
+                    # contain many mimeparts that refer to the same data.
+                    # Flanker creates all mimeparts and as we load their body
+                    # prop, we can end up with thousands of similar mimeparts,
+                    # each having their own copy of the actual data they refer too.
+                    # Here is the part where Mimepart.stream keeps the reference
+                    # to copy of body, and never removes it later. If the body is
+                    # 16MB of each mimepart, and we have thousands of mimeparts,
+                    # we run out of memory.
+                    # https://github.com/closeio/flanker/blob/master/flanker/mime/message/part.py#L75
+                    # By resetting the body attribute we remove the data copy for that mimepart.
+                    # In case the prop was accessed after that, it would just be
+                    # loaded again. So we trade cpu for memory
+                    mimepart.body = None
```

Comment:
Ideally flanker would not keep a list, but just a stream of mimeparts, and thus remove them once they are processed in the loop. But that would be a bigger change imo and in flanker

Reply:
Should this be `mimepart._body = None` so we don't overwrite the property itself?

Reply:
This looks good, but maybe we should just say `mimepart._body = None` so we don't set `_body_changed` (since we don't actually change the body), just in case.

Reply:
That makes sense. `    mimepart._container._body = None`